flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -63,6 +89,10 @@
  */
 public abstract class ClusterEntrypoint implements FatalErrorHandler {
 
+	public static final ConfigOption<String> EXECUTION_MODE = ConfigOptions
+		.key("cluster.execution-mode")
```

Comment:
Is it the first time that `ConfigOptions` are used for something that the user should/cannot not configure?

Reply:
There are other configuration parameters as well like `local.number-taskmanager`. In the old Yarn code, we used environment variables to pass certain settings. I think the configuration is a better place. However, I agree that this might be a bit confusing. I'll add the prefix `internal` to the key.